Unsafe cast in notification listener dispatch.

This code does not do a proper check before casting:
```
if (message instanceof ObjectStatus) {
  listener.objectStatusNotification((ObjectStatus) message);
} else {
  listener.otherEventNotification((OtherEventNotifications) message);
}	
```
This is causing issues in the openhab binding:
```
2017-07-05 20:28:11.709 [ERROR] [om.digitaldan.jomnilinkII.Connection] - Notifcation Handler Caught Exception
java.lang.ClassCastException: com.digitaldan.jomnilinkII.MessageTypes.AudioSourceStatus cannot be cast to com.digitaldan.jomnilinkII.MessageTypes.OtherEventNotifications
        at com.digitaldan.jomnilinkII.Connection$NotificationHandler.run(Connection.java:844)[263:org.openhab.binding.omnilink:2.2.0.201707050940]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,1 +1,2 @@
 build/
+target/

--- a/src/main/java/com/digitaldan/jomnilinkII/Connection.java
+++ b/src/main/java/com/digitaldan/jomnilinkII/Connection.java
@@ -840,8 +840,10 @@ public class Connection extends Thread {
 						for (NotificationListener listener : listeners) {
 							if (message instanceof ObjectStatus) {
 								listener.objectStatusNotification((ObjectStatus) message);
-							} else {
+							} else if (message instanceof OtherEventNotifications){
 								listener.otherEventNotification((OtherEventNotifications) message);
+							} else{
+								logger.debug("Unhandled notficiation message: {}", message);
 							}
 						}
 					}

--- a/src/main/java/com/digitaldan/jomnilinkII/Connection.java
+++ b/src/main/java/com/digitaldan/jomnilinkII/Connection.java
@@ -839,7 +839,7 @@ public class Connection extends Thread {
 					} else {
 						for (NotificationListener listener : listeners) {
 							if (message instanceof ObjectStatus) {
-								listener.objectStausNotification((ObjectStatus) message);
+								listener.objectStatusNotification((ObjectStatus) message);
 							} else {
 								listener.otherEventNotification((OtherEventNotifications) message);
 							}
@@ -847,7 +847,7 @@ public class Connection extends Thread {
 					}
 				} catch (Throwable t) {
 					//Catch all exceptions to prevent notification thread from dying.
-					logger.error("Notifcation Handler Caught Exception", t);
+					logger.error("Notification Handler Caught Exception", t);
 					t.printStackTrace();
 				}
 			}

--- a/src/main/java/com/digitaldan/jomnilinkII/NotificationListener.java
+++ b/src/main/java/com/digitaldan/jomnilinkII/NotificationListener.java
@@ -4,7 +4,7 @@ import com.digitaldan.jomnilinkII.MessageTypes.ObjectStatus;
 import com.digitaldan.jomnilinkII.MessageTypes.OtherEventNotifications;
 
 public interface NotificationListener {
-	public void objectStausNotification(ObjectStatus status);
+	public void objectStatusNotification(ObjectStatus status);
 
 	public void otherEventNotification(OtherEventNotifications other);
 }

--- a/src/main/java/com/digitaldan/jomnilinkII/examples/Main.java
+++ b/src/main/java/com/digitaldan/jomnilinkII/examples/Main.java
@@ -50,7 +50,7 @@ public class Main {
 			c.debug = true;
 			c.addNotificationListener(new NotificationListener() {
 				@Override
-				public void objectStausNotification(ObjectStatus s) {
+				public void objectStatusNotification(ObjectStatus s) {
 					switch (s.getStatusType()) {
 					case Message.OBJ_TYPE_AREA:
 						logger.info("STATUS_AREA changed");

